libs/application/templates/estate/src/fields/TextFieldsRepeater/index.tsx

```diff
@@ -97,6 +106,14 @@ export const TextFieldsRepeater: FC<FieldBaseProps<Answers> & Props> = ({
                       currency={field.currency}
                       readOnly={field.readOnly}
                       type={field.type}
+                      onChange={(e) => {
+                        setIndex(fieldIndex)
+                        if (field.id === 'rateOfExchange') {
+                          setRateOfExchange(Number(e.target.value))
+                        } else if (field.id === 'faceValue') {
+                          setFaceValue(Number(e.target.value))
```

Comment:
Are these Integer-Only fields?
If so, consider using `Number.parseInt(e.target.value, 10)` instead.
Otherwise, this is fine (I think) if fractions are fine.